src/led_ble/led_ble.py

```diff
@@ -466,8 +507,10 @@ def _notification_handler(self, _sender: int, data: bytearray) -> None:
         if not self._resolve_protocol_event.is_set():
             self._resolve_protocol_event.set()
             self._model_data = get_model(model_num)
-            self._set_protocol(self._model_data.protocol_for_version_num(version))
-
+            if self._is_hello_fairy():
+                self._protocol = ProtocolFairy()
+            else:
+                self._set_protocol(self._model_data.protocol_for_version_num(version))
```

Comment:
`protocol_for_version_num` should handle this instead of setting it directly to the new protocol

Reply:
LEDBLEModel doesn't know what kind of device it is.  I worked around this by calling hello fairy "version 0" but it may actually have its own versions.  One solution would be to have an object for devices that implements protocol_for_version_number [see my other comment](https://github.com/Bluetooth-Devices/led-ble/pull/56#discussion_r1895006329).  Another would be to map the hello fairy version string to an enum and just make it disjoint from the flux_led devices (and any other devices that get implemented someday).

Reply:
I remembered that wrong, it's model that I set to 0... So this method will get a protocol name "Fairy" but `PROTOCOL_NAME_TO_CLS.get(protocol)` isn't going to work because the mapping is specified in flux_led.  I can push the `if is_hello_fairy` down into `_set_protocol` though.